src/openssl_wrap.cpp

```diff
@@ -363,7 +363,7 @@ int _BIO_puts(BIO* bio, std::string buf)
 
 int _BIO_write(BIO* bio, const std::vector<uint8_t> &data)
 {
-    if (data.size() > std::numeric_limits<int>::max()) {
+    if (data.size() > static_cast<std::decay_t<decltype(data)>::size_type>(std::numeric_limits<int>::max())) {
```

Comment:
Why this complicated? Readability counts, this is no template library. std::vector<uint8_t>::size_type is easily understandable to everyone if we want to be super future-proof. Otherwise, casting to size_t is probably already good enough.

Reply:
Sorry, I did this ;-)

Reply:
Done